File: src/main/java/com/somefrills/features/misc/glowmob/chestui/UIAddon.java
package com.somefrills.features.misc.glowmob.chestui;

import net.minecraft.inventory.Inventory;
import net.minecraft.item.ItemStack;
import java.util.List;

public interface UIAddon {
    // Allows the addon to filter or slice the item list
    void processItems(ChestUI ui, List<ItemStack> items);

    // Allows the addon to place its own buttons (arrows, compass, etc.)
    void drawDecoration(ChestUI ui, Inventory inventory);

    // Allows the addon to intercept clicks
    boolean onClick(ChestUI ui, ItemStack stack, String name, int button);
}

File: src/main/kotlin/com/somefrills/features/misc/glowmob/chestui/UIAddon.kt
package com.somefrills.features.misc.glowmob.chestui

import net.minecraft.world.Container
import net.minecraft.world.item.ItemStack


interface UIAddon {
    // Allows the addon to filter or slice the item list
    fun processItems(ui: ChestUI, items: MutableList<ItemStack>)

    // Allows the addon to place its own buttons (arrows, compass, etc.)
    fun drawDecoration(ui: ChestUI, inventory: Container)

    // Allows the addon to intercept clicks
    fun onClick(ui: ChestUI, stack: ItemStack, name: String, button: Int): Boolean
}